collector/lib/Logging.h

```diff
@@ -127,11 +127,14 @@ class LogMessage {
 
 #define CLOG(lvl) CLOG_IF(true, lvl)
 
-#define CLOG_THROTTLED_IF(cond, lvl, interval)                                          \
-  static std::chrono::steady_clock::time_point _clog_lastlog_##__LINE__;                \
-  if (collector::logging::CheckLogLevel(collector::logging::LogLevel::lvl) && (cond) && \
-      (std::chrono::steady_clock::now() - _clog_lastlog_##__LINE__ >= interval))        \
-  _clog_lastlog_##__LINE__ = std::chrono::steady_clock::now(),                          \
+#define CLOG_THROTTLED_IF(cond, lvl, interval)                                           \
+  static bool _clog_firstlog_##__LINE__ = true;                                          \
+  static std::chrono::steady_clock::time_point _clog_lastlog_##__LINE__;                 \
+  if (_clog_firstlog_##__LINE__ ||                                                       \
+      (collector::logging::CheckLogLevel(collector::logging::LogLevel::lvl) && (cond) && \
+       (std::chrono::steady_clock::now() - _clog_lastlog_##__LINE__ >= interval)))       \
+  _clog_lastlog_##__LINE__ = std::chrono::steady_clock::now(),                           \
+  _clog_firstlog_##__LINE__ = false,                                                     \
```

Comment:
I think it might be simpler to initialize `lastlog` to `std::chrono::steady_clock::now()` to avoid the first log line being missed

Reply:
I think the original idea was that having only `_clog_lastlog_##__LINE__` and the condition `std::chrono::steady_clock::now() - _clog_lastlog_##__LINE__ >= interval` means nothing will be logged right after start (which is rather dubious case, but still could happen).